src/routes/delegates/v2/delegates.v2.service.ts

```diff
@@ -59,4 +63,56 @@ export class DelegatesV2Service {
       label: args.createDelegateDto.label,
     });
   }
+
+  async deleteDelegate(args: {
+    chainId: string;
+    delegateAddress: `0x${string}`;
+    deleteDelegateV2Dto: DeleteDelegateV2Dto;
+  }): Promise<unknown> {
+    const { deleteDelegateV2Dto } = args;
+    let { delegator } = deleteDelegateV2Dto;
+
+    if (!delegator) {
+      if (!deleteDelegateV2Dto.safe) {
+        // Note: this point shouldn't be reached, schema validation should cover it
+        throw Error(
+          'Either safe or delegator should be non-null when deleting a delegate',
+        );
+      }
+      delegator = await this._getDelegatorForDelegateAndSafe(
+        args.chainId,
+        args.delegateAddress,
+        deleteDelegateV2Dto.safe,
+      );
+    }
+
+    return await this.repository.deleteDelegate({
+      chainId: args.chainId,
+      delegate: args.delegateAddress,
+      delegator,
+      safeAddress: deleteDelegateV2Dto.safe,
+      signature: deleteDelegateV2Dto.signature,
+    });
+  }
+
+  private async _getDelegatorForDelegateAndSafe(
+    chainId: string,
+    delegate: `0x${string}`,
+    safeAddress: `0x${string}`,
+  ): Promise<`0x${string}`> {
+    const response = await this.repository.getDelegates({
+      chainId,
+      delegate,
+      safeAddress,
+    });
+
+    if (!response.results.length) {
+      this.loggingService.warn(
+        `Delegator for delegate ${delegate} and Safe ${safeAddress} not found`,
+      );
+      throw new NotFoundException();
+    }
+
+    return response.results[0].delegator;
+  }
```

Comment:
Are you trying to create one function for both the deletion of delegates and Safe delegates?

I think it would be simpler to have a function for each. Individual schemas/types should also result in no need for unnecessary `throw` conditions. What do you think?

Reply:
Yes, I'm trying to unify both endpoints, as the deletion of "delegates" and "safe delegates" is essentially the same action in the end, the only difference is the client sending different parameters. Having one endpoint would be more in line with the Transaction Service API.

Do you think maintaining the two endpoints is more desirable? The clients will need to make changes either way, as the signatures needs to be calculated in a different manner.

Reply:
I assumed that we would continue to have both, apologies. I think aligning more with the Transaction Service makes sense, considering the clients will need to make respective adjustments as well.